db/migrate/20241204085540_remap_to_fa6_icon_names.rb

```diff
@@ -0,0 +1,794 @@
+          when /^fab fa-/
+            prefix = "fab"
+            lookup_name = icon_name.sub(/^fab fa-/, "")
+          when /^far fa-/
+            prefix = "far"
+            lookup_name = icon_name.sub(/^far fa-/, "")
+          when /^fas fa-/
+            lookup_name = icon_name.sub(/^fas fa-/, "")
+          when /^fa-/
+            lookup_name = icon_name.sub(/^fa-/, "")
+          else
+            lookup_name = icon_name
+          end
+
+          new_icon_name = FA5_REMAPS[lookup_name] || lookup_name
+          new_icon_name = "#{prefix}-#{new_icon_name}" if prefix
+          new_icon_name
+        end
+        .join("|")
+
+    return if new_setting_value == original_setting_value
+
+    execute(
+      "UPDATE site_settings SET value = '#{new_setting_value}' WHERE name = '#{SVG_ICON_SUBSET_SETTING_NAME}'",
+    )
+  end
+
+  def migrate_icon_names
+    mappings = FA5_REMAPS.map { |from, to| "('#{from}', '#{to}')" }.join(", ")
+
+    tables_to_update = {
+      groups: "flair_icon",
+      post_action_types: "icon",
+      badges: "icon",
+      sidebar_urls: "icon",
+      directory_columns: "icon",
+    }
+
+    tables_to_update.each do |table_name, column_name|
+      execute <<~SQL
+        WITH remaps AS (
+          SELECT from_icon, to_icon FROM (VALUES #{mappings}) AS mapping(from_icon, to_icon)
+        )
+        UPDATE #{table_name}
+        SET #{column_name} =
+          CASE
+            WHEN #{column_name} LIKE 'fab-%' THEN CONCAT('fab-', remaps.to_icon)
+            WHEN #{column_name} LIKE 'far-%' THEN CONCAT('far-', remaps.to_icon)
+            WHEN #{column_name} LIKE 'fab fa-%' THEN CONCAT('fab-', remaps.to_icon)
+            WHEN #{column_name} LIKE 'far fa-%' THEN CONCAT('far-', remaps.to_icon)
+            ELSE remaps.to_icon
+          END
+        FROM remaps
+        WHERE #{column_name} = remaps.from_icon
+          OR #{column_name} = CONCAT('fa-', remaps.from_icon)
+          OR #{column_name} = CONCAT('far-', remaps.from_icon)
+          OR #{column_name} = CONCAT('fab-', remaps.from_icon)
+          OR #{column_name} = CONCAT('far fa-', remaps.from_icon)
+          OR #{column_name} = CONCAT('fab fa-', remaps.from_icon)
+          OR #{column_name} = CONCAT('fas fa-', remaps.from_icon);
+      SQL
+
+      execute <<~SQL
+        UPDATE #{table_name}
+        SET #{column_name} =
+        CASE
+          WHEN #{column_name} LIKE 'fas fa-%' THEN SUBSTRING(#{column_name} FROM 8)
+          WHEN #{column_name} LIKE 'far fa-%' THEN CONCAT('far-', SUBSTRING(#{column_name} FROM 8))
+          WHEN #{column_name} LIKE 'fab fa-%' THEN CONCAT('fab-', SUBSTRING(#{column_name} FROM 8))
+          WHEN #{column_name} LIKE 'fa-%' THEN SUBSTRING(#{column_name} FROM 4)
+          ELSE #{column_name}
+        END
+        WHERE #{column_name} LIKE 'fa-%'
+          OR #{column_name} LIKE 'far fa-%'
+          OR #{column_name} LIKE 'fab fa-%'
+          OR #{column_name} LIKE 'fas fa-%';
+      SQL
+    end
```

Comment:
This is the real diff from the previous migration. Theoretically the line `ELSE #{column_name}` should never be reached but this acts as a failsafe to set to the current value instead of NULL which would be very bad (data loss).